controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -42,14 +42,25 @@ const (
 type RosaControlPlaneSpec struct { //nolint: maligned
 	// Cluster name must be valid DNS-1035 label, so it must consist of lower case alphanumeric
 	// characters or '-', start with an alphabetic character, end with an alphanumeric character
-	// and have a max length of 15 characters.
+	// and have a max length of 54 characters.
 	//
+	// +immutable
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="rosaClusterName is immutable"
-	// +kubebuilder:validation:MaxLength:=15
+	// +kubebuilder:validation:MaxLength:=54
 	// +kubebuilder:validation:Pattern:=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
-	// +immutable
 	RosaClusterName string `json:"rosaClusterName"`
 
+	// Domain prefix must be valid DNS-1035 label, so it must consist of lower case alphanumeric
+	// characters or '-', start with an alphabetic character, end with an alphanumeric character
+	// and have a max length of 15 characters.
+	//
+	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="domainPrefix is immutable"
+	// +kubebuilder:validation:MaxLength:=15
+	// +kubebuilder:validation:Pattern:=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
+	// +optional
+	DomainPrefix string `json:"domainPrefix"`
```

Comment:
can we have a more descriptive comment here of what this field does instead. 
`DomainPrefix is an optional unique domain prefix of the cluster. This will be used when generating a sub-domain for your cluster on openshiftapps.com`

Reply:
update it